src/JsonSchema/Constraints/UndefinedConstraint.php

```diff
@@ -66,7 +66,8 @@ public function validateTypes(&$value, $schema = null, JsonPointer $path, $i = n
         }
 
         // check object
-        if ($this->getTypeCheck()->isObject($value)) {
+        if (TypeCheck\LooseTypeCheck::isObject($value)) { // Fixes failing assoc tests for default values - currently investigating
+        //if ($this->getTypeCheck()->isObject($value)) {  // to find the root cause of this, noting all other assoc tests pass.
```

Comment:
Yup. Now you're hitting the next problem, which is that in `LooseTypeCheck::propertySet` you are doing a write to $value, which is passed by value, so the write doesn't stick. You need to either pass by reference, or pass in an object. It was working in your old code before because when you derferred to `testValidCases`, that was converting to object. 

Hope that all makes sense. I'm off to bed, but I can help more tomorrow morning if you leave some comments. G'nite!